dev/tests/functional/tests/app/Magento/ConfigurableProduct/Test/Constraint/AssertConfigurableProductInCategory.php

```diff
@@ -28,12 +27,19 @@ protected function assertPrice(FixtureInterface $product, CatalogCategoryView $c
         $priceBlock = $catalogCategoryView->getListProductBlock()->getProductItem($product)->getPriceBlock();
         $priceData = $product->getDataFieldConfig('price')['source']->getPriceData();
         $price = isset($priceData['category_price']) ? $priceData['category_price'] : $product->getPrice();
-        \PHPUnit\Framework\Assert::assertEquals(
-            number_format($price, 2, '.', ''),
-            $priceBlock->isOldPriceVisible() ? $priceBlock->getOldPrice() : $priceBlock->getPrice(),
-            'Product regular price on category page is not correct.'
-        );
-
+        if ($priceBlock->isNormalPriceVisible()) {
+            \PHPUnit\Framework\Assert::assertEquals(
+                number_format($price, 2, '.', ''),
+                $priceBlock->getNormalPrice(),
+                'Product normal price on category page is not correct.'
```

Comment:
@p-bystritsky how is normal price different from regular price?

Reply:
@orlangur actual configurable product price on category page is located in span with 'normal_price' class.